### PR TITLE
bazel: detect go standard library deprecation errors

### LINKED ISSUE #84877: bazel: `nogo` does not seem to catch all deprecation errors
Found in [TC run](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/5824100?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true) for #84590 at `a8c77d7ad18cadba1310c04e526cd845e2a090c6`:

```
        pkg/util/netutil/net.go:140:52: ne.Temporary has been deprecated since Go 1.18 because it shouldn't be used: Temporary errors are not well-defined. Most "temporary" errors are timeouts, and the few exceptions are surprising. Do not use this method.  (SA1019)
    lint_test.go:1765:
        pkg/util/netutil/net.go:173:11: netError.Temporary has been deprecated since Go 1.18 because it shouldn't be used: Temporary errors are not well-defined. Most "temporary" errors are timeouts, and the few exceptions are surprising. Do not use this method.  (SA1019)
```

`nogo` did not raise this.

Epic CRDB-8349

Jira issue: CRDB-17922

### PR DESCRIPTION
**Pull request by healthy-pod**
Previously, `staticcheck` only raised deprecation errors
from non-stdlib packages because the analysis pass used
by the deprecation analyzer is created by `nogo` which doesn't
analyze the standard library packages.

This code change patches `staticcheck` to let it detect deprecation
errors from the standard library and raise them. It also removes two
uses of `tls.Config.PreferServerCipherSuites` because it's deprecated
since Go 1.18 and is a legacy field that is ignored and has no effect.

Closes #84877

Release justification: Non-production code change
Release note: None

### CODE REVIEW

**Review by rickystewart — APPROVED** [review]

**Comment by rickystewart** [comment]
@healthy-pod This may warrant a bug report upstream. It's known that stdlib packages don't always have analysis result and I think this check should still work even in that scenario. Best case scenario we might be able to remove this custom patch in the future.

**Comment by healthy-pod** [comment]
bors r=rickystewart

**Comment by knz** [comment]
This is missing the CLA approval. Need to get that done before the PR can be processed.

**Comment by healthy-pod** [comment]
> This is missing the CLA approval. Need to get that done before the PR can be processed.

Right thanks for pointing that out. I merged PRs before though so trying to find out why `license/cla` was not reported this time.

**Comment by knz** [comment]
Sometimes the webhook doesn't fire. `git commit --amend --no-edit` and re-push usually tickles it enough.

**Comment by knz** [comment]
you can issue the bors command now. no need to wait for CI

**Comment by healthy-pod** [comment]
bors r=rickystewart

**Pull request merged**
